ocpp-1-6-api-adapter/src/main/kotlin/com/izivia/ocpp/adapter16/Ocpp16CSApiAdapter.kt

```diff
@@ -263,7 +263,13 @@ class Ocpp16CSApiAdapter(
         req: SetChargingProfileReq
     ): OperationExecution<SetChargingProfileReq, SetChargingProfileResp> {
         val mapper: SetChargingProfileMapper = Mappers.getMapper(SetChargingProfileMapper::class.java)
-        val response = csApi.setChargingProfile(meta, mapper.coreToGenReq(req))
+        val transactionId = req.csChargingProfiles.transactionId
+            ?.let { transactionIds.getLocalIdByTransactionId(it) }
+            ?.localId
+        val response = csApi.setChargingProfile(
+            meta,
+            mapper.coreToGenReq(req)
+                .let { it.copy(chargingProfile = it.chargingProfile.copy(transactionId = transactionId)) })
```

Comment:
I think this would deserve a unit test to check the behavior and illustrate the use of the local transaction ids.